packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Arrange from 'terra-arrange';
+import styles from './SectionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Text to be displayed on the SectionHeader.
+   */
+  title: PropTypes.node.isRequired,
+  /**
+   * Callback function triggered when the accordion icon is clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Whether the accordion icon should be displayed in its 'open' or 'closed' position.
+   */
+  isOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  onClick: undefined,
+  isOpen: false,
+};
+
+const SectionHeader = ({
+  title,
+  onClick,
+  isOpen,
+  ...customProps
+}) => {
+  if ((process.env.NODE_ENV !== 'production') && (!onClick && isOpen)) {
+    // eslint-disable-next-line no-console
+    console.warn('\'isOpen\' are intended to be used only when \'onClick\' is provided.');
+  }
+
+  const attributes = Object.assign({}, customProps);
+
+  if (onClick) {
+    attributes.tabIndex = '0';
+  }
+
+  const iconClassNames = cx([
+    'accordion-icon',
+    { 'is-open': onClick && isOpen },
```

Comment:
IMO the is-open class should not be tied the `onClick` prop and only the `isOpen` prop.

Reply:
`onClick` is the determinant of whether or not an accordion icon can be displayed. By doing `onClick && isOpen`, I short-circuit the logical evaluation and don't care about whether or not the user wants the icon to be displayed in its 'open' state or not, as there is no icon to be displayed.